js/sites.js

```diff
@@ -197,12 +198,12 @@
       // first check for end time, as it may modify the link
       this._checkLinkEndTime();
       // setup display variables
-      //let enhanced = Services.prefs.getBoolPref("browser.newtabpage.enhanced")
-      //                && DirectoryLinksProvider.getEnhancedLink(this.link);
-      let enhanced = gNewTab.enhanced;
+      let enhanced = this.link;
       let url = this.url;
-      let title = enhanced && enhanced.title ? enhanced.title :
-        this.link.type === "history" ? this.link.baseDomain :
+      let title = enhanced && enhanced.title &&
```

Comment:
Sorry, but is it possible to rewrite this (even though it's not your code)? nested ternary statements are super confusing :(